forge-game/src/main/java/forge/game/staticability/StaticAbilityDisableTriggers.java

```diff
@@ -90,25 +89,40 @@ public static boolean isDisabled(final StaticAbility stAb, final TriggerType tri
                 }
             }
         } else if (triggerType.equals(TriggerType.ChangesZoneAll)) {
-            final String destination = stAb.getParamOrDefault("Destination", null);
+            final String origin = stAb.getParam("Origin");
+            final String destination = stAb.getParam("Destination");
             final CardZoneTable table = (CardZoneTable) runParams.get(AbilityKey.Cards);
 
-            // find out if any other cards would still trigger it
-            boolean found = false;
+            // purge all forbidden causes from table
+            CardZoneTable filtered = new CardZoneTable();
+            boolean possiblyDisabled = false;
             for (Cell<ZoneType, ZoneType, CardCollection> cell : table.cellSet()) {
                 // this currently assumes the table will not contain multiple destinations
                 // however with some effects (e.g. Goblin Welder) that should indeed be the case
                 // once Forge handles that correctly this section needs to account for that
                 // (by doing a closer check of the triggered ability first)
-                if (cell.getColumnKey() != ZoneType.valueOf(destination)) {
-                    found = true;
-                } else if (Iterables.any(cell.getValue(),
-                        Predicates.not(CardPredicates.isType(stAb.getParam("ValidCause"))))) {
-                    found = true;
+                CardCollection changers = cell.getValue();
+                if ((origin == null || cell.getRowKey() == ZoneType.valueOf(origin)) &&
+                (destination == null || cell.getColumnKey() == ZoneType.valueOf(destination))) {
+                    changers = CardLists.filter(changers, Predicates.not(CardPredicates.isType(stAb.getParam("ValidCause"))));
+                    // the static will match some of the causes
+                    if (changers.size() < cell.getValue().size()) {
+                        possiblyDisabled = true;
```

Comment:
If this is the only place that changes possiblyDisabled to true, could we remove the boolean and just return false here?

Reply:
Well that might be safe only if there's no more than a single cell to check

That shouldn't be true in all cases (and if Forge doesn't mirror that yet completely in the future it should)

Reply:
Ah, that makes sense.